applier/applier_test.go

```diff
@@ -500,6 +506,246 @@ func TestApplier_ApplyLayers(t *testing.T) {
 			},
 			wantErr: "no packages detected",
 		},
+		{
+			name: "happy path with custom resources",
+			args: args{
+				imageID: "sha256:4791503518dff090d6a82f7a5c1fd71c41146920e2562fb64308e17ab6834b7e",
+				layerIDs: []string{
+					"sha256:932da51564135c98a49a34a193d6cd363d8fa4184d957fde16c9d8527b3f3b02",
+					"sha256:dffd9992ca398466a663c87c92cfea2a2db0ae0cf33fcb99da60eec52addbfc5",
+				},
+			},
+			getLayerExpectations: []cache.LocalArtifactCacheGetBlobExpectation{
+				{
+					Args: cache.LocalArtifactCacheGetBlobArgs{
+						BlobID: "sha256:932da51564135c98a49a34a193d6cd363d8fa4184d957fde16c9d8527b3f3b02",
+					},
+					Returns: cache.LocalArtifactCacheGetBlobReturns{
+						BlobInfo: types.BlobInfo{
+							SchemaVersion: 1,
+							Digest:        "sha256:932da51564135c98a49a34a193d6cd363d8fa4184d957fde16c9d8527b3f3b02",
+							DiffID:        "sha256:a187dde48cd289ac374ad8539930628314bc581a481cdb41409c9289419ddb72",
+							PackageInfos: []types.PackageInfo{
+								{
+									FilePath: "var/lib/dpkg/status.d/tzdata",
+									Packages: []types.Package{
+										{
+											Name:       "tzdata",
+											Version:    "2019a-0+deb9u1",
+											SrcName:    "tzdata",
+											SrcVersion: "2019a-0+deb9u1",
+										},
+									},
+								},
+							},
+							CustomResources: []types.CustomResource{
+								{
+									Type:     "type-A",
+									FilePath: "var/lib/dpkg/status.d/tzdata",
+									Data: dummyData{
+										data: "Common Package type-A var/lib/dpkg/status.d/tzdata",
+									},
+								},
+								{
+									Type:     "type-B",
+									FilePath: "var/lib/dpkg/status.d/tzdata",
+									Data: dummyData{
+										data: "Common Package type-B var/lib/dpkg/status.d/tzdata",
+									},
```

Comment:
What is this for? We have `type-C` and it seems to be enough.

Reply:
This case was added for case, if same resource on same layer is added by different analyzers, they should be equivalently merged. Tesecase check for intralayer merging. 
Type-C case is specific to same resource, on **different** layer by different analyzer.

Reply:
>if same resource on same layer is added by different analyzers

Type-C seems to satisfy the above condition.

We want to test the following cases.

1. The same resource on the same layer by different analyzers
2. The same resource on the different layers by the same analyzer

The pair of Type-A and Type-C seems to cover both. Or, am I missing something? Can Type-B test something new?

Reply:
Above mentioned testcases are satisfied by Type-A and Type-C. 
But following testcase is missed. 
Consider a single layer with same resource(1 as Type-A and 1 as Type-C) but detected by different analyzer. If we have bug introduced in it merging this data. The above testcase may mask this failure, as the data will be overridden by merge on the next layer.

Reply:
Above mentioned testcases are satisfied by Type-A and Type-C. 
But following testcase is missed. 
Consider a single layer with same resource(1 as Type-A and 1 as Type-C) but detected by different analyzer. If we have bug introduced in it merging this data(intralayer). The above testcase may mask this failure, as the data will be overridden by merge on the next layer.

Reply:
What if we move Type-A to the upper layer?

- upper layer
  - foo.txt
    - Type-A
    - Type-C
- lower layer
  - foo.txt
    - Type-C

Reply:
okey. I will update it accordingly